src/lbfgs.jl

```diff
@@ -172,29 +209,35 @@ function SolverCore.solve!(
     ∇f .= ∇ft
 
     ∇fNorm = nrm2(n, ∇f)
-    iter = iter + 1
 
+    set_solution!(stats, x)
```

Comment:
This is costly to do at each iteration. The current iterate is already in the solver object. We should only store it in `stats` at the end.

Reply:
Thanks for the feedback. I also applied this to R2.